includes/polling/ports.inc.php

```diff
@@ -174,45 +174,85 @@
     'dot3StatsDuplexStatus',
 ];
 
+// Query known ports and mapping table in order of discovery to make sure
+// the latest discoverd/polled port is in the mapping tables.
+$ports_mapped = get_ports_mapped($device['device_id'], true);
+$ports = $ports_mapped['ports'];
+
+//
+// Rename any old RRD files still named after the previous ifIndex based naming schema.
+foreach ($ports_mapped['maps']['ifIndex'] as $ifIndex => $port_id) {
+    foreach (array ('', '-adsl', '-dot3') as $suffix) {
+        $old_rrd_name = "port-$ifIndex$suffix";
+        $new_rrd_name = getPortRrdName($port_id, ltrim($suffix, '-'));
+
+        rrd_file_rename($device, $old_rrd_name, $new_rrd_name);
+    }
+}
+
 echo 'Caching Oids: ';
-$port_stats = array();
-$data       = array();
+$port_stats = [];
 
 if ($device['os'] === 'f5' && (version_compare($device['version'], '11.2.0', '>=') && version_compare($device['version'], '11.7', '<'))) {
     require_once 'ports/f5.inc.php';
 } else {
     if ($config['polling']['selected_ports'] === true || $config['os'][$device['os']]['polling']['selected_ports'] === true || $device['attribs']['selected_ports'] == 'true') {
-        echo('Select ports polling');
-        $lports = dbFetchRows("SELECT * FROM `ports` where `device_id` = ? AND `deleted` = 0 AND `disabled` = 0", array($device['device_id']));
-        foreach ($lports as $lport) {
-            if (is_port_valid($lport, $device)) {
-                $i = $lport['ifIndex'];
-                $base_oids = implode(".$i ", $table_base_oids) . ".$i";
-                $data = snmp_get_multi($device, $base_oids, '-OQUst', 'IF-MIB', null, $data);
-                if ($lport['ifAdminStatus_prev'] === 'down' && $data[$i]['ifAdminStatus'] === 'down') {
-                    echo 'port is still admin down';
-                } elseif ($lport['ifOperStatus_prev'] === 'down' && $data[$i]['ifOperStatus'] === 'down') {
-                    echo 'port is still down';
+        echo 'Selected ports polling ';
+
+        // remove the deleted and disabled ports and mark them skipped
+        $polled_ports = array_filter($ports, function ($port) use ($ports) {
+            $ports[$port['ifIndex']]['skipped'] = true;
+            return $port['deleted'] || $port['disabled'];
+        });
+
+        // if less than 5 ports or less than 10% of the total ports are skipped, walk the base oids instead of get
+        $polled_port_count = count($polled_ports);
+        $total_port_count = count($ports);
+        $walk_base = $total_port_count - $polled_port_count < 5 || $polled_port_count / $total_port_count > 0.9 ;
+
+        foreach ($table_base_oids as $oid) {
```

Comment:
Is this not supposed to check `$walk_base`?

Reply:
Yeah, I think it is.